### PR TITLE
Backport of 3.x flash messages #5823

### PR DESCRIPTION
**Pull request by tanuck**
Had a quick attempt to make the Flash component & helper from 3.x compatible for 2.7. 

Few things I was unsure of though:
- The message data is stored under the Flash session index, as in 3.x. Should it use the Message index like the Session component?
- Seen as it is not the only option for flash message creation, I haven't included example element files in the app/View/Element directory. Thoughts?

### CODE REVIEW

**Comment by markstory** [comment]
Looks good to me :+1:

**Comment by tanuck** [comment]
:+1: Think there some phpdoc's that need updating that I missed.

File: lib/Cake/Controller/Component/FlashComponent.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+ * @param string $name Element name to use.
+ * @param array $args Parameters to pass when calling `FlashComponent::set()`.
+ * @return void
+ * @throws InternalErrorException If missing the flash message.
+ */
+	public function __call($name, $args) {
+		$options = array('element' => Inflector::underscore($name));
+
+		if (count($args) < 1) {
+			throw new InternalErrorException(__('Flash message missing.'));
```
Comment (dereuromark) [inline lib/Cake/Controller/Component/FlashComponent.php]: Shouldnt that have the "cake_core" domain using __d()?

Comment (tanuck) [inline lib/Cake/Controller/Component/FlashComponent.php]: I agree. Think it probably makes more sense to remove it rather than use __d()

**Pull request merged**